packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -333,11 +333,10 @@ export async function fetchContent({ syncToken, pluginConfig, reporter }) {
   } finally {
     // Fix output when there was no new data in Contentful
     if (
-      currentSyncData?.entries.length +
-        currentSyncData?.assets.length +
-        currentSyncData?.deletedEntries.length +
-        currentSyncData?.deletedAssets.length ===
-      0
+      !currentSyncData?.entries.length &&
+      !currentSyncData?.assets.length &&
+      !currentSyncData?.deletedEntries.length &&
+      !currentSyncData?.deletedAssets.length
```

Comment:
Thsi was giving TS errors, because it was trying to perform arithmetic when the value is possibly null (thanks to the optional chain)